### PR TITLE
Add flag version

### PR DESCRIPTION
**Pull request by cwdsuzhou**
Support print version use `--version`

### CODE REVIEW

**Comment by pohly** [comment]
This looks useful overall.

I'm less sure about printing the full path of the binary (`os.Args[0]`). Perhaps just print the base name?

**Comment by cwdsuzhou** [comment]
> This looks useful overall.
> 
> I'm less sure about printing the full path of the binary (`os.Args[0]`). Perhaps just print the base name?

It prints the base name. I do not think it is necessary to print the full path. What we concern is the version of the binary overall

**Comment by pohly** [comment]
Weidong Cai <notifications@github.com> writes:

>> This looks useful overall.
>> 
>> I'm less sure about printing the full path of the binary (`os.Args[0]`). Perhaps just print the base name?
>
> It prints the base name.

That depends on how you invoke it:
```
$ /nvme/gopath/src/github.com/kubernetes-csi/csi-driver-host-path/bin/hostpathplugin --version
/nvme/gopath/src/github.com/kubernetes-csi/csi-driver-host-path/bin/hostpathplugin 6584b2ae0cc4cd2966299913b81ee070f67b009a
```

Including the path is unusual. Other tools don't do that, for example:
```
$ /home/pohly/bin/kind --version
kind version 0.2.0-alpha
```

**Comment by cwdsuzhou** [comment]
Ok，I will do that, thank you!

**Comment by cwdsuzhou** [comment]
> Weidong Cai <notifications@github.com> writes:
> > This looks useful overall. > > I'm less sure about printing the full path of the binary (`os.Args[0]`). Perhaps just print the base name? It prints the base name.
> That depends on how you invoke it: ``` $ /nvme/gopath/src/github.com/kubernetes-csi/csi-driver-host-path/bin/hostpathplugin --version /nvme/gopath/src/github.com/kubernetes-csi/csi-driver-host-path/bin/hostpathplugin [6584b2a](https://github.com/kubernetes-csi/csi-driver-host-path/commit/6584b2ae0cc4cd2966299913b81ee070f67b009a) ``` Including the path is unusual. Other tools don't do that, for example: ``` $ /home/pohly/bin/kind --version kind version 0.2.0-alpha ```

@pohly I have done that.

File: cmd/hostpathplugin/main.go [hunk 1]
```diff
@@ -29,14 +30,23 @@ func init() {
...
+	nodeID      = flag.String("nodeid", "", "node id")
+	showVersion = flag.Bool("version", false, "Show version.")
+	version     = ""
 )
 
 func main() {
 	flag.Parse()
 
+	if *showVersion {
+		paths := strings.Split(os.Args[0], "/")
```
Comment (pohly) [inline cmd/hostpathplugin/main.go]: Better use https://godoc.org/path#Base.

Comment (cwdsuzhou) [reply cmd/hostpathplugin/main.go]: Thank you for your advise, I have changed it.

Comment (cwdsuzhou) [reply cmd/hostpathplugin/main.go]: @pohly please review it again, thank you!

File: cmd/hostpathplugin/main.go [hunk 2]
```diff
@@ -29,14 +30,22 @@ func init() {
...
 
 var (
-	endpoint   = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
-	driverName = flag.String("drivername", "csi-hostpath", "name of the driver")
-	nodeID     = flag.String("nodeid", "", "node id")
+	endpoint    = flag.String("endpoint", "unix://tmp/csi.sock", "CSI endpoint")
+	driverName  = flag.String("drivername", "csi-hostpath", "name of the driver")
+	nodeID      = flag.String("nodeid", "", "node id")
+	showVersion = flag.Bool("version", false, "Show version.")
+	version     = ""
```
Comment (msau42) [inline cmd/hostpathplugin/main.go]: Where is this set?  How do we correlate version with git tags and container image versions?

Comment (cwdsuzhou) [reply cmd/hostpathplugin/main.go]: > Where is this set? How do we correlate version with git tags and container image versions?

It is set dynamically when building in the release-tool, `--version` only helps print it. How It is set is decided by the release-tool and  it equals the git commit revision by default. If you want to release a version, you can tag it eg. v.1.0.0, then, image  version would be v1.0.0 and the `--version` prints the result beginning with v.1.0.0, e.g, v1.0-0-ge18d8b63-dirty. For more details, you can refer to [https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/release-tools/build.make](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/release-tools/build.make)

Comment (msau42) [reply cmd/hostpathplugin/main.go]: Thanks for the clarification! Can you add a comment to this saying that it's set by the build process?

Comment (cwdsuzhou) [reply cmd/hostpathplugin/main.go]: > Thanks for the clarification! Can you add a comment to this saying that it's set by the build process?

Thanks for your review! I have  add a comment to it.

File: cmd/hostpathplugin/main.go [hunk 3]
```diff
@@ -34,7 +34,8 @@ var (
 	driverName  = flag.String("drivername", "csi-hostpath", "name of the driver")
 	nodeID      = flag.String("nodeid", "", "node id")
 	showVersion = flag.Bool("version", false, "Show version.")
-	version     = ""
+	//Set by the build process
```
Comment (msau42) [inline cmd/hostpathplugin/main.go]: just one nit to add a space after "//".  And also squash the commits.

Comment (cwdsuzhou) [reply cmd/hostpathplugin/main.go]: Thank you , please review the code again.

**Pull request merged**